meson.build

```diff
@@ -55,6 +55,27 @@ warning_c_flags = [
 ]
 
 
+if get_option('enable-fp16')
+   arch = target_machine.cpu_family()
+   if get_option('platform') == 'android'
+     add_project_arguments('-mfp16-format=ieee', language: ['c', 'cpp'])
+     extra_defines += '-DENABLE_FP16=1'
+     extra_defines += '-DUSE__FP16=1'
+   elif arch == 'aarch64' or arch =='arm'
+     extra_defines += '-DENABLE_FP16=1'
```

Comment:
As we are allowing FP16 to "arm", Does the above set flag specific to aarch64 cause any problem?

if get_option('enable-fp16')
   flags += '-march=armv8.2-a+fp16 -mfpu=neon-fp16 -mfloat-abi=softfp'
endif